incubator/gocd/templates/gocd-server-deployment.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "gocd.fullname" . }}-server
+  labels:
+    app: {{ template "gocd.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+    component: {{ template "gocd.name" . }}-server
+spec:
+  {{- if le .Values.server.replicaCount 1.0 }}
+  replicas: {{ .Values.server.replicaCount }}
+  {{- end }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "gocd.name" . }}
+        release: {{ .Release.Name | quote }}
+        component: {{ template "gocd.name" . }}-server
+    spec:
+      {{- if .Values.server.persistence.enabled }}
+      volumes:
+        - name: goserver-vol
+          persistentVolumeClaim:
+            claimName: {{ template "gocd.fullname" . }}
+      {{- end }}
+      containers:
+        - name: {{ template "gocd.name" . }}-server
+          {{- if .Values.server.image.tag }}
+          image: "{{ .Values.server.image.repository }}:{{ .Values.server.image.tag }}"
+          {{- else }}
+          image: "{{ .Values.server.image.repository }}:v{{ default .Chart.AppVersion .Values.app.version }}"
```

Comment:
I don't think we should have an additional `app.version`. It's redundant. You should just use the image tag instead.